Library/Homebrew/cleanup.rb

```diff
@@ -160,11 +160,18 @@ def self.install_formula_clean!(f)
       cleanup = Cleanup.new
       if cleanup.periodic_clean_due?
         cleanup.periodic_clean!
-      elsif f.latest_version_installed?
+      elsif f.latest_version_installed? && !cleanup.skip_clean_formula?(f)
         cleanup.cleanup_formula(f)
       end
     end
 
+    def skip_clean_formula?(f)
+      return false if Homebrew::EnvConfig.no_cleanup_formulae.blank?
+
+      skip_clean_formulae = Homebrew::EnvConfig.no_cleanup_formulae.split(",")
```

Comment:
I wonder if it's worth adding this behaviour into `EnvConfig` itself under the `type`? We also seem to alternate between comma and space separated on our configuration; might be worth doubling down on one?

Reply:
We actually also have an option that takes a path to a file that specifies the list. We should probably only pick one approach, but this will require a deprecation cycle.

Reply:
Or: we support both space-separated and comma-separated in the cases where we can use both (e.g. environment variables), document consistently and only use comma-separated in the cases where only it works (e.g. arguments)

Reply:
Yep, fine with me. I'll try to look into it.